Aquarius/Src/Aquarius/Sound/Devices.cpp

```diff
@@ -45,15 +45,27 @@ namespace Aquarius {
             }
         }
 
+        Device* Device::device = nullptr;
+
         Device * Device::get()
         {
-            static Device* device = new Device();
+            if (device == nullptr)
+            {
+                device = new Device();
+            }
+
+            else
+            {
+                device = nullptr;
+            }
```

Comment:
Don't need this, this would actually cause a memory leak as you are removing the only reference to a heap object without deleting it first. The above if statement is sufficient, it just creates the device if it doesn't already exist, and if not it just returns it